### PR TITLE
Refactor of EventHubStreamProvider to better support plugable CacheDataAdapter

### PR DESCRIPTION
**Pull request by jason-bragg**
To test the ability for users to provide their own cache data adapter to the EventHubStreamProvider, I attempted to add a test that uses a custom data adapter to produce a single stream per partition.  This attempt exposed flaws in the design.

This PR contains refactors to the EventHubStreamProvider to support this scenario, as well as the Stream per partition test.

### CODE REVIEW

File: src/Orleans/Streams/Core/StreamIdentity.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿
+using System;
+
+namespace Orleans.Streams
+{
+    public class StreamIdentity : IStreamIdentity
```
Comment (sergeybykov) [inline src/Orleans/Streams/Core/StreamIdentity.cs]: Please add XML comments for new public types.

File: src/OrleansProviders/Streams/Common/PooledCache/StreamPosition.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿
+using System;
+using Orleans.Streams;
+
+namespace Orleans.Providers.Streams.Common
+{
+    public class StreamPosition
```
Comment (sergeybykov) [inline src/OrleansProviders/Streams/Common/PooledCache/StreamPosition.cs]: Another public class with no XML docs.

File: src/OrleansServiceBus/Providers/Streams/EventHub/EventHubBatchContainer.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@
 namespace Orleans.ServiceBus.Providers
 {
     [Serializable]
-    internal class EventHubBatchContainer : IBatchContainer
+    public class EventHubBatchContainer : IBatchContainer
```
Comment (sergeybykov) [inline src/OrleansServiceBus/Providers/Streams/EventHub/EventHubBatchContainer.cs]: Another one.

File: src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs [hunk 4]
```diff
@@ -12,14 +12,14 @@ namespace Orleans.ServiceBus.Providers
...
-    internal struct CachedEventHubMessage
+    public struct CachedEventHubMessage
     {
         public Guid StreamGuid;
         public long SequenceNumber;
         public ArraySegment<byte> Segment;
     }
 
-    internal class EventHubDataComparer : ICacheDataComparer<CachedEventHubMessage>
+    public class EventHubDataComparer : ICacheDataComparer<CachedEventHubMessage>
```
Comment (sergeybykov) [inline src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs]: And one more.

File: src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs [hunk 5]
```diff
@@ -42,7 +42,7 @@ public int Compare(CachedEventHubMessage cachedMessage, IStreamIdentity streamId
         }
     }
 
-    internal class EventHubDataAdapter : ICacheDataAdapter<EventData, CachedEventHubMessage>
+    public class EventHubDataAdapter : ICacheDataAdapter<EventData, CachedEventHubMessage>
```
Comment (sergeybykov) [inline src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs]: And one more.

File: src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCache.cs [hunk 6]
```diff
@@ -6,7 +6,7 @@
 
 namespace Orleans.ServiceBus.Providers
 {
-    public abstract class EventHubQueueCache<TCachedMessage> : IEventHubQueueCache
+    public abstract class EventHubQueueCacheBase<TCachedMessage> : IEventHubQueueCache
```
Comment (sergeybykov) [inline src/OrleansServiceBus/Providers/Streams/EventHub/EventHubQueueCache.cs]: Can we avoid 'Base'? Such a name suffix is frown upon these days. And it's another public class with no docs.

**Comment by jason-bragg** [comment]
Comments addressed, re-based, squashed.

**Pull request merged**